File: legacy/backend/app/domain/models/rule.py
from dataclasses import dataclass, field
from typing import Any


@dataclass(frozen=True)
class RuleDefinition:
    id: str
    scope: str
    name: str
    enabled: bool
    priority: int
    description: str = ""
    condition: dict[str, Any] = field(default_factory=dict)
    effects: list[dict[str, Any]] = field(default_factory=list)

File: src-tauri/src/models/rule.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RuleDefinition {
    pub id: String,
    pub scope: String,
    pub name: String,
    pub enabled: bool,
    pub priority: i32,
    pub description: String,
    pub condition: serde_json::Value,
    pub effects: Vec<serde_json::Value>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RuleCreateRequest {
    pub scope: String,
    pub name: String,
    pub enabled: bool,
    pub priority: i32,
    pub description: String,
    pub condition: serde_json::Value,
    pub effects: Vec<serde_json::Value>,
}
